src/compile/model.ts

```diff
@@ -280,10 +280,10 @@ export abstract class Model {
     return assembleLegends(this);
   }
 
-  public assembleGroup(signals: VgSignal[] = []) {
+  public assembleGroup(mixins: {signals?: VgSignal[], scales: VgScale[]}) {
```

Comment:
I just found a way to make this part less convoluted! 
https://github.com/vega/vega-lite/pull/2698/commits/fbfc544565b0a248704bac95970d4734d7ada971